In [1]:
import numpy as np
from scipy import stats

### Задача 1
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

Т.к. нам известна дисперсия, то используем следующую статистику $T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}.$

Остальные параметры из условия задачи равны: $a = 17$, $\alpha = 0.025$, $\sigma = 2$, $\overline{X} = 17.5$

Определим значение правосторонней критической области. Для этого воспользуемся функцией norm.ppf из модуля stats.

In [2]:
alpha = 0.025

stats.norm.ppf(1 - alpha / 2)

2.241402727604947

Получается что наша правосторонняя критическая область равна $\Omega_\alpha = (2.2414, \infty)$.
Теперь подставим наши значения и проверим попало ли значение статистики $T$ в критическую область.
$T(X) = \dfrac{17.5 - 17}{2 / 10} = 5 \cdot 0.5 = 2.5.$

Значение $2.5$ попадает в нашу критическую область. Таким образом гипотеза о том что шарики для подшипников имеют средний диаметр 17 мм неверна.

### Задача 2
Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания $a$ с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

Зафиксируем значения из условия: $\overline{X} = 80$, $n = 256$, $p = 0.95$, $\sigma = 16$. Т.к. по факту нам известна дисперсия, то воспользуемся следующим доверительным интервалом:
$
P \left(
    \overline{X} - t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq a \leq 
    \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$

$p = 1 - \alpha$, отсюда $\alpha = 1 - p = 1 - 0.95 = 0.05$. Мы знаем что для стандартного распределения и $\alpha = 0.05$ значение $t_{1 - \alpha/2}$ будет равно $2$.

In [7]:
mean = 80
n = 256
t = 2
std = 16
(mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n))

(78.0, 82.0)

### Задача 3
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

Предположим что утверждение продавца неверно.
Как обычно зафиксируем значения: $\overline{X} = 200$, $n = 10$, $\alpha = 0.01$. Т.к. дисперсия нам неизвестна, то воспользуемся t-распределением Стьюдента. Найдем недостающее значение несмещенной оценки для среднеквадратического отклонения и среднее выборки.

In [4]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mean = samples.mean()
std = samples.std(ddof=1)
mean, std

(198.5, 4.453463071962462)

Для построения критической области найдем нужный нам квантиль $t_{1 - \alpha/2, \: n - 1}$

In [5]:
alpha = 0.01
n = 10
t = stats.t.ppf(1 - alpha / 2, df=n - 1)
t

3.2498355440153697

Критическая область в нашем случае будет $\Omega_\alpha = \left( -\infty, -3.2498 \right) \cup \left( 3.2498, \infty \right).$

Теперь соберем все вместе и посмотрим попадает ли наше значение статистики в критическую область.

In [6]:
X = 200
T = (X - mean) / (std / np.sqrt(n))
T

1.0651074037450896

Т.к. значение статистики $T$ не попало в критическую область, то утверждение продавца истинно.